script.js

```diff
@@ -1,96 +1,229 @@
-
-  //creating eposides and display them with their names and summary.
-  for (let i = 0 ; i < episodeList.length ; i++ ){
-    divEle = document.createElement('div');
-   divEle.id = 'episode'
-   mainEle.appendChild(divEle)
-   let titleEle = document.createElement('h1')
-   titleEle.id = 'title'
-   divEle.appendChild(titleEle)
-   titleEle.innerText = nameOfEpisodes(episodeList[i])
-   let EpisodeImage = document.createElement('img')
-      EpisodeImage.src = episodeList[i].image.medium;
-   divEle.appendChild(EpisodeImage);
-   let paraEle = document.createElement('div')
-   paraEle.id = 'para'
-   paraEle.innerHTML = episodeList[i].summary
-   divEle.appendChild(paraEle)
+   //////////-------list of Shows profiles--------------->
+function listOfShowsProfiles(orderedShows){
+  displayingNumOfEpisodes(orderedShows, orderedShows, 'SHOW/S')
 
+ ///////////////////////////////////////////////////////////
+ orderedShows.forEach(show=> {
+  let showProfile = document.createElement('div')
+  showProfile.id = 'show-profile'
+  let inShowProfile = document.createElement('div')
+  inShowProfile.id = 'inShowProfile'
+  let title = document.createElement('h1')
+  title.id = 'show-title'
+  title.innerText = show.name
+  let showImage = document.createElement('img')
+  if(show.image){
+    showImage.src = show.image.medium
+  } else {
+    showImage.src = 'https://encrypted-tbn0.gstatic.com/images?q=tbn%3AANd9GcRqEWgS0uxxEYJ0PsOb2OgwyWvC0Gjp8NUdPw&usqp=CAU'
+  }
+  let showSummary = document.createElement('p')
+  showSummary.innerHTML = show.summary
+  showImage.className = 'showImage'
+  showImage.value = show.id
+  let showDiscription = document.createElement('div')
+  let showRate = document.createElement('p')
+  let showGener = document.createElement('p')
+  let showStatus = document.createElement('p')
+  let showRuntime = document.createElement('p')
+let replaceComma = show.genres.toString().replace(/,/g, "|")
+  showRate.innerHTML = 'Rated: ' + show.rating.average
+  showGener.innerHTML = 'Gener: ' + replaceComma
+  showStatus.innerHTML = 'Status: ' + show.status
+  showRuntime.innerHTML = 'Runtime: ' + show.runtime
+ let sticker = document.createElement('div')
+ sticker.id = 'showSticker'
+ sticker.appendChild(title)
+ sticker.appendChild(showImage)
+ inShowProfile.appendChild(sticker)
+  inShowProfile.appendChild(showSummary)
+  showDiscription.appendChild(showRate)
+  showDiscription.appendChild( showGener)
+  showDiscription.appendChild(showStatus )
+  showDiscription.appendChild(showRuntime)
+  inShowProfile.appendChild(showDiscription)
+  showProfile.appendChild(inShowProfile)
+  mainEle.appendChild(showProfile)
+  showImage.addEventListener('click', function (){
+
+    let url = `https://api.tvmaze.com/shows/${showImage.value}/episodes`
+    fetch(url)
+    .then((response)=> response.json())
+    .then((data)=> {
+     let allEpisodes = data
+        makePageForEpisodes(allEpisodes);
+    })
+    .catch(error=>console.log(error))
```

Comment:
Great job including error handling. This is really important when you or other developers are debugging, and it can provide us with a clearer idea of what's wrong instead of just getting a default error message 💯